pkg/beacon/relay/node.go

```diff
@@ -157,6 +158,61 @@ func (n *Node) ForwardSignatureShares(groupPublicKeyBytes []byte) {
 	n.netProvider.BroadcastChannelForwarderFor(name)
 }
 
+// ResumeSigningIfEligible enables a client to rejoin the ongoing signing process
+// after it was crashed or restarted and if it belongs to the signing group.
+func (n *Node) ResumeSigningIfEligible(
+	relayChain relayChain.Interface,
+	signing chain.Signing,
+) {
+	isEntryInProgress, err := relayChain.IsEntryInProgress()
+	if err != nil {
+		logger.Errorf(
+			"failed checking if an entry is in progress: [%v]",
+			err,
+		)
+		return
+	}
+
+	if isEntryInProgress {
+		previousEntry, err := relayChain.CurrentRequestPreviousEntry()
+		if err != nil {
+			logger.Errorf(
+				"failed to get a previous entry for current request: [%v]",
+				err,
+			)
+			return
+		}
+		entryStartBlock, err := relayChain.CurrentRequestStartBlock()
+		if err != nil {
+			logger.Errorf(
+				"failed to get a current relay request entry start block: [%v]",
+				err,
+			)
+			return
+		}
+		groupPublicKey, err := relayChain.CurrentRequestGroupPublicKey()
+		if err != nil {
+			logger.Errorf(
+				"failed to get current relay request's group public key: [%v]",
+				err,
+			)
+			return
+		}
+
+		logger.Infof(
+			"client is atempting to rejoin the current signing processs [0x%x]",
```

Comment:
`attempting` and `process`

Reply:
Also, let's just say `attempting to rejoin the current signing process...`. I don't think we use `client is...` anywhere else, we just say `leaving message loop...`, `accepting signature share...`, `rejecting signature share...` and so on.

Reply:
sure, done